frappe/contacts/doctype/address/address.py

```diff
@@ -81,16 +81,14 @@ def has_common_link(self, doc):
 @frappe.whitelist()
 def get_default_address(doctype, name, sort_key='is_primary_address'):
 	'''Returns default Address name for the given doctype, name'''
-	out = frappe.db.sql('''select
-			parent,
-			(select `{0}` from tabAddress a where a.name=dl.parent) as `{0}`
-		from
-			`tabDynamic Link` dl
-		where
-			link_doctype=%s and
-			link_name=%s and
-			parenttype = "Address"
-		'''.format(sort_key), (doctype, name))
+	out = frappe.db.sql(""" SELECT
+			addr.name, addr.`{sort_key}`
+		FROM
+			`tabAddress` addr, `tabDynamic Link` dl
+		WHERE
+			dl.parent = addr.name and dl.link_doctype = %s and
+			dl.link_name = %s and ifnull(addr.disabled, 0) = 0
+		""".format(sort_key = sort_key), (doctype, name))
```

Comment:
check for injection, since your are getting input from the public (validate it must be `is_primary_address` or whatever key you are expecting.

Reply:
Fixed!